packages/components/pie-cookie-banner/test/helpers/page-object/pie-cookie-banner.page.ts

```diff
@@ -0,0 +1,138 @@
+import { type Locator, type Page } from '@playwright/test';
+import { BasePage } from '@justeattakeaway/pie-webc-testing/src/helpers/page-object/base-page.ts';
+import {
+    cookieBanner,
+} from './selectors.ts';
+import {
+    PreferenceIds,
+} from '../../../src/defs.ts';
+
+const getPreferenceItemSelector = (id: PreferenceIds) => `#${id} [data-test-id="switch-component"]`;
+
+export class PieCookieBannerComponent extends BasePage {
+    readonly componentLocator: Locator;
+    readonly descriptionLocator: Locator;
+    readonly acceptAllButtonLocator: Locator;
+    readonly necessaryOnlyButtonLocator: Locator;
+    readonly managePrefsLinkLocator: Locator;
+    readonly bodyAcceptAllLinkLocator: Locator;
+    readonly bodyNecessaryOnlyLinkLocator;
+    readonly bodyManagePreferencesLinkLocator: Locator;
+    readonly bodyCookieStatementLinkLocator: Locator;
+    readonly bodyCookieTechnologiesLinkLocator: Locator;
+    readonly modalDescriptionLocator: Locator;
+
+    constructor (page: Page) {
+        super(page);
+        this.componentLocator = page.getByTestId(cookieBanner.selectors.container.dataTestId);
+        this.descriptionLocator = page.getByTestId(cookieBanner.selectors.description.dataTestId);
+        this.acceptAllButtonLocator = page.getByTestId(cookieBanner.selectors.acceptAllButton.dataTestId);
+        this.necessaryOnlyButtonLocator = page.getByTestId(cookieBanner.selectors.necessaryOnlyButton.dataTestId);
+        this.managePrefsLinkLocator = page.getByTestId(cookieBanner.selectors.managePreferencesLink.dataTestId);
+        this.bodyAcceptAllLinkLocator = page.getByTestId(cookieBanner.selectors.bodyAcceptAllLink.dataTestId);
+        this.bodyNecessaryOnlyLinkLocator = page.getByTestId(cookieBanner.selectors.bodyNecessaryOnlyLink.dataTestId);
+        this.bodyManagePreferencesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyManagePreferencesLink.dataTestId);
+        this.bodyCookieStatementLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieStatementLink.dataTestId);
+        this.bodyCookieTechnologiesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieTechnologiesLink.dataTestId);
+        this.modalDescriptionLocator = page.getByTestId(cookieBanner.selectors.modalDescription.dataTestId);
+    }
+
+    async isCookieBannerVisible () {
```

Comment:
NIT: I would add typing to these methods. I.e. `: boolean` to show what type they return